source/game_sa/CarAI.cpp

```diff
@@ -95,7 +191,19 @@ eCarMission CCarAI::FindPoliceBoatMissionForWantedLevel() {
 // rtype eCarMission ?
 // 0x41C9D0
 int8 CCarAI::FindPoliceCarMissionForWantedLevel() {
-    return plugin::CallAndReturn<int8, 0x41C9D0>();
+    double probability = CGeneral::GetRandomNumberInRange(0.0, 1.0);
+    switch (FindPlayerWanted()->m_nWantedLevel) {
+    case 2:
+        return probability < 0.75 ? MISSION_BLOCKPLAYER_FARAWAY : MISSION_RAMPLAYER_FARAWAY;
+    case 3:
+        return probability < 0.50 ? MISSION_BLOCKPLAYER_FARAWAY : MISSION_RAMPLAYER_FARAWAY;
+    case 4:
+    case 5:
+    case 6:
+        return probability < 0.25 ? MISSION_BLOCKPLAYER_FARAWAY : MISSION_RAMPLAYER_FARAWAY;
+    default:
+        return MISSION_BLOCKPLAYER_FARAWAY;
+    }
```

Comment:
This can be refactored further but it's not important for now.
```suggestion
    switch (FindPlayerWanted()->m_nWantedLevel) {
    case 2:
        return CGeneral::RandomBool(0.75f) ? MISSION_BLOCKPLAYER_FARAWAY : MISSION_RAMPLAYER_FARAWAY;
    case 3:
        return CGeneral::RandomBool(0.50f) ? MISSION_BLOCKPLAYER_FARAWAY : MISSION_RAMPLAYER_FARAWAY;
    case 4:
    case 5:
    case 6:
        return CGeneral::RandomBool(0.25f) ? MISSION_BLOCKPLAYER_FARAWAY : MISSION_RAMPLAYER_FARAWAY;
    default:
        return MISSION_BLOCKPLAYER_FARAWAY;
    }
```

Reply:
It's just the chance changing every time.
Could even add a lambda above:
```cpp
const auto Chooce = [](float blockProbability) {  return CGeneral::RandomBool(blockProbability) ? MISSION_BLOCKPLAYER_FARAWAY : MISSION_RAMPLAYER_FARAWAY; };
```

Reply:
Changed it to randombool

Reply:
I just remembered though, `RandomBool` uses [0, 100] interval, so don't forget to fix that.